src/claimtrie.cpp

```diff
@@ -137,6 +149,75 @@ void CClaimTrieNode::reorderClaims(supportMapEntryType& supports)
     std::make_heap(claims.begin(), claims.end());
 }
 
+CClaimTrie* CClaimTrie::enableNormalizationFork(bool enable, int nHeight)
+{
+    LogPrintf("%s: Normalize Name Fork is %s\n",
+        __func__, (enable ? "ENABLED" : "DISABLED"));
+
+    if (enable) {
+        std::vector<namedNodeType> nodes = flattenTrie();
+        LogPrintf("Pref-normalization fork claim trie nodes = %lu\n", nodes.size());
+        // Debug-only
+        /* for (std::vector<namedNodeType>::iterator it = nodes.begin(); it != nodes.end(); ++it) */
+        /* { */
+        /*     if (!it->second.claims.empty()) */
+        /*         LogPrintf("[---] Claim name: \"%s\" (hash %s)\n", it->first.c_str(), it->second.hash.GetHex().c_str()); */
+        /*     else */
+        /*         LogPrintf("[EMPTY] Claim name: \"%s\" (hash %s)\n", it->first.c_str(), it->second.hash.GetHex().c_str()); */
+        /* } */
+
+        if (normalizationForkStateInstance.nonNormalizedTrie == NULL) {
+            normalizationForkStateInstance.nonNormalizedTrie = this;
+            normalizationForkStateInstance.normalizedTrie = new CClaimTrie(false, false, true);
```

Comment:
Leaking this is not a problem right now, but if we use placement or in-class operator new (that i've plans to do it) will be.
```
static CClaimTrie normalizedTrie(false, false, true);
normalizationForkStateInstance.normalizedTrie = &normalizedTrie;
```
It'll be better

Reply:
Mine concern is that operator delete isn't called, it can do many other things than to free memory.

Reply:
Understood, but it needs to remain in memory and intact past the fork.  A future release will remove it (by deleting it) when it's clear that we can.  If you're generally pointing out that the single pointer is leaked, it's not particularly helpful because we know there are leaks in the claimtrie already (and this one isn't one in particular to be concerned about, IMO).  The proposed suggestion doesn't change that either.  Upstream uses managed pointers, and while our first update to it will not, it will be much easier to transition after that update.  I was originally planning to do that before the upstream release, but it makes sense to do it after since we can skip boost pointers altogether and just use std ones.

Reply:
Fixed in current push.